app/services/image_upload_service.rb

```diff
@@ -31,8 +31,8 @@ def upload_to_asset_manager(image)
     AssetManagerService.new.upload_bytes(image, image.cropped_bytes)
   end
 
-  def filename
-    file.respond_to?(:original_filename) ? file.original_filename : File.basename(file)
+  def filename(document)
+    @filename ||= ImageFilenameService.new(document).call(file.original_filename, mime_type)
```

Comment:
Should the changes to this service be in a separate commit like: "Use new ImageFilenameService to generate the filename"?